webnn/resources/test_data/linear.json

```diff
@@ -0,0 +1,611 @@
+      "inputs": {
+        "x": {
+         "shape": [2, 2, 2, 3],
+          "data": [
+            -1.1225161816883542,
+            -6.605223195131225,
+            -1.9555539248683687,
+            -4.598548331438281,
+            4.234208072658522,
+            3.097542121236385,
+            3.7465923873418348,
+            -4.487029772637743,
+            6.407402114982858,
+            -4.354544013606705,
+            -5.8190925441357955,
+            3.7214346452179328,
+            -6.330114110872294,
+            8.580595073862398,
+            -6.764922344398516,
+            6.433565829855674,
+            -9.708685944356201,
+            2.6431380232113746,
+            5.214088864824586,
+            9.658617012096975,
+            -8.72174939837989,
+            -0.4533396492565025,
+            9.992619953447026,
+            -6.469674921617137
+          ],
+          "type": "float32"
+        }
+      },
+      "options": {
+        "alpha": 7.398793812746618
+      },
+      "expected": {
+        "name": "output",
+        "shape": [2, 2, 2, 3],
+        "data": [
+          -8.305265426635742,
+          -48.87068176269531,
+          -14.46873950958252,
+          -34.023712158203125,
+          31.328031539916992,
+          22.918073654174805,
+          27.72026252746582,
+          -33.198604583740234,
+          47.407047271728516,
+          -32.2183723449707,
+          -43.05426788330078,
+          27.53412628173828,
+          -46.835205078125,
+          63.486053466796875,
+          -50.05226516723633,
+          47.600624084472656,
+          -71.83256530761719,
+          19.556032180786133,
+          38.57796859741211,
+          71.46211242675781,
+          -64.53042602539062,
+          -3.3541665077209473,
+          73.9333267211914,
+          -47.86779022216797
+        ],
+        "type": "float32"
+      }
+    },
+    {
+      "name": "linear float32 4D tensor specified positive options.beta and default options.alpha",
+      "inputs": {
+        "x": {
+         "shape": [2, 2, 2, 3],
+          "data": [
+            -1.1225161816883542,
+            -6.605223195131225,
+            -1.9555539248683687,
+            -4.598548331438281,
+            4.234208072658522,
+            3.097542121236385,
+            3.7465923873418348,
+            -4.487029772637743,
+            6.407402114982858,
+            -4.354544013606705,
+            -5.8190925441357955,
+            3.7214346452179328,
+            -6.330114110872294,
+            8.580595073862398,
+            -6.764922344398516,
+            6.433565829855674,
+            -9.708685944356201,
+            2.6431380232113746,
+            5.214088864824586,
+            9.658617012096975,
+            -8.72174939837989,
+            -0.4533396492565025,
+            9.992619953447026,
+            -6.469674921617137
+          ],
+          "type": "float32"
+        }
+      },
+      "options": {
+        "beta": 5.919095653700928
```

Comment:
Beware adding a mix of negative and positive numbers for beta, as you could get catastrophic cancellation. Adding positive beta to positive inputs or negative beta to negative inputs should ensure your ULP tolerance is valid.

Same with next two cases after this.

Reply:
Also done to update test data to avoid catastrophic cancellation, please take another look, thanks! @fdwr